scio-core/src/main/scala/com/spotify/scio/io/dynamic/syntax/SCollectionSyntax.scala

```diff
@@ -87,8 +86,14 @@ final class DynamicSpecificRecordSCollectionOps[T <: SpecificRecord](
         .withCodec(codec)
         .withMetadata(nm)
       val write =
-        writeDynamic(path, numShards, suffix, destinationFn, tempDirectory)
-          .via(sink)
+        writeDynamic(
+          path = path,
+          destinationFn = destinationFn,
+          numShards = numShards,
+          prefix = prefix,
+          suffix = suffix,
+          tempDirectory = tempDirectory
```

Comment:
Why the provide the parameter names explicitly here? Did you mix up prefix/suffix?

Reply:
Yes, Since API is not too strongly types, using named parameter helped to make sure I did not mess up with parameter ordering